test/grpc/route_guide/route-guide-eventuals-server.h

```diff
@@ -0,0 +1,163 @@
+#include "test/grpc/route_guide/make.h"
+#include "test/grpc/route_guide/route_guide.eventuals.h"
+#include "test/grpc/route_guide/route_guide.grpc.pb.h"
+
+////////////////////////////////////////////////////////////////////////
+
+inline float ConvertToRadians(float num) {
+  return num * 3.1415926 / 180;
+}
+
+////////////////////////////////////////////////////////////////////////
+
+// The formula is based on http://mathforum.org/library/drmath/view/51879.html
+inline float GetDistance(
+    const routeguide::Point& start,
+    const routeguide::Point& end) {
+  const float kCoordFactor = 10000000.0;
+  float lat_1 = start.latitude() / kCoordFactor;
+  float lat_2 = end.latitude() / kCoordFactor;
+  float lon_1 = start.longitude() / kCoordFactor;
+  float lon_2 = end.longitude() / kCoordFactor;
+  float lat_rad_1 = ConvertToRadians(lat_1);
+  float lat_rad_2 = ConvertToRadians(lat_2);
+  float delta_lat_rad = ConvertToRadians(lat_2 - lat_1);
+  float delta_lon_rad = ConvertToRadians(lon_2 - lon_1);
+
+  float a = pow(sin(delta_lat_rad / 2), 2)
+      + cos(lat_rad_1) * cos(lat_rad_2) * pow(sin(delta_lon_rad / 2), 2);
+  float c = 2 * atan2(sqrt(a), sqrt(1 - a));
+  int R = 6371000; // metres
+
+  return R * c;
+}
+
+////////////////////////////////////////////////////////////////////////
+
+inline std::string GetFeatureName(
+    const routeguide::Point& point,
+    const std::vector<routeguide::Feature>& feature_list) {
+  for (const routeguide::Feature& f : feature_list) {
+    if (f.location().latitude() == point.latitude()
+        && f.location().longitude() == point.longitude()) {
+      return f.name();
+    }
+  }
+  return "";
+}
+
+////////////////////////////////////////////////////////////////////////
+
+class RouteGuideImpl final
+  : public routeguide::eventuals::RouteGuide::Service<RouteGuideImpl>,
+    public eventuals::Synchronizable {
+ public:
+  RouteGuideImpl()
+    : feature_list_([]() {
+        std::string db = routeguide::GetDbFileContent();
+        std::vector<routeguide::Feature> temp;
+        routeguide::ParseDb(db, &temp);
+        return temp;
+      }()) {}
+
+  routeguide::Feature GetFeature(
+      grpc::ServerContext* context,
+      routeguide::Point&& point) {
+    routeguide::Feature feature;
+    feature.set_name(GetFeatureName(point, feature_list_));
+    feature.mutable_location()->CopyFrom(point);
+    return feature;
+  }
+
+  auto ListFeatures(
+      grpc::ServerContext* context,
+      routeguide::Rectangle&& rectangle) {
+    auto lo = rectangle.lo();
+    auto hi = rectangle.hi();
+    long left = std::min(lo.longitude(), hi.longitude());
+    long right = std::max(lo.longitude(), hi.longitude());
+    long top = std::max(lo.latitude(), hi.latitude());
+    long bottom = std::min(lo.latitude(), hi.latitude());
+
+    return eventuals::Iterate(feature_list_)
+        >> eventuals::Filter([left, right, top, bottom](
+                                 const routeguide::Feature& f) {
```

Comment:
What if you wrap the entire lambda? Will that be less than 80 characters?
```suggestion
        >> eventuals::Filter(
            [left, right, top, bottom](const routeguide::Feature& f) {
```